### ISSUE TITLE
[ESC + .] does not work in Warp

### ISSUE THREAD

**Issue by huxiang** [warpdotdev/Warp#1597]
### Describe the bug

[ECS + .] in other terminal programs is "last word of last command": https://tutonics.com/2012/05/handy-bash-shortcuts.html

it does not work in Wrap

### To Reproduce

Steps to reproduce:
1. Open Warp
2. Run command `ls /tmp`
3. Type `find ` and then press [ECS+.]

### Expected behaviour

It becomes `find /tmp`

### Operating System

MacOS

### OS Version

12.4

### Shell Version

5.8.1

### Warp Version

v0.2022.07.04.09.08.stable_01

### Additional context

BTW, I cannot copy Warp version info from UI...

### Does this block you from using Warp daily?

Yes

**Comment by rushi** [comment]
I came here to report this as well. This option is basically `<meta> + .` and when you select "Left Option Key is Meta" or "Right Optoin Key is Meta" in Warp, doing `⌥.` should return the last argument of the previous command, but instead it outputs `≥`

**Comment by rushi** [comment]
Quick workaround - You can type `$_` 

```bash
mkdir foo
cd $_
```

**Comment by elviskahoro** [comment]
Will cirlce back with yall once there are updates. But I don't see any coming anytime soon. :(

**Comment by jab** [comment]
Just heard about Warp and was excited to try it right away, then hit this within my first minute of trying it. Would it be worth adding a warning to the relevant part of the settings UI, which currently makes it seem like option-as-meta is going to Just Work? 

![Screen Shot 2022-09-29 at 3 51 52 PM](https://user-images.githubusercontent.com/64992/193128547-7b8a0ac1-cfde-4c3b-930b-9fb8fc9d5a80.png)

**Comment by jab** [comment]
Also could a Warp team member please update this ticket's title to fix the typos, namely "ESC" rather than "ECS" and "Warp" rather than "Wrap"?

**Comment by jab** [comment]
> Quick workaround - You can type $_

This does not work in fish shell.

**Comment by warpdotdev-dx** [comment]
This should have been fixed today!

--- original message ---
On September 29, 2022 at 3:55 PM EDT ***@***.*** wrote:



> Quick workaround - You can type $_


This does not work in fish shell.

—

Reply to this email directly, view it on GitHub, or unsubscribe.

You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

-- 
You received this message because you are subscribed to the Google Groups "github" group.
To unsubscribe from this group and stop receiving emails from it, send an email to ***@***.***
--- end of original message ---

**Comment by eleeaz95** [comment]
ESC key has other uses apart from ESC+.
Why set up a new "meta" key for that?
I think it would be better to use ESC in the same way as other shells.

**Comment by warpdotdev-dx** [comment]
Can you elaborate some on what the ideal product experience would be for you!

**Comment by eleeaz95** [comment]
Well, I frequently use the [ESC+.] on other terminals to bring the last word of the last command (usually directories). For me will be a good experience to have that, or to have the option to set ESC as a meta key.  ESC has other uses aswell, you can see some examples here: https://tutonics.com/2012/05/handy-bash-shortcuts.html

**Comment by jab** [comment]
> This should have been fixed today!

Nice, thanks for the quick fix! Any idea how soon before we can update to a release with the fix? Checked https://github.com/warpdotdev/Warp/releases and the latest release is from April.

@elviskahoro, thanks for fixing one of the typos in the issue title. The other typo still needs to be fixed: Wrap -> Warp

**Comment by elviskahoro** [comment]
@jab Warp should auto-update. We don't actively maintain the GitHub releases section. You can also install Warp with brew

**Comment by kr3cj** [comment]
I'm running version `v0.2023.03.07.08.02.stable_03` of warp which is latest stable at the time of writing. I have unchecked both `Left Option key is Meta` and `Right Option key is Meta` as mentioned above. My shell is `zsh`. However, when trying to use `Esc + .`, it still won't work for me in warp. It just inserts the `period` as if I wasn't pushing `Esc`. It works okay in `terminal` and `iTerm2`. Any tips?

**Comment by lubokkanev** [comment]
Doesn't work for me either.

**Comment by xidiot** [comment]
@elviskahoro , this is still a BUG. Should we ope a new issue or?

**Comment by dannyneira** [comment]
Hey Folks, we've implemented this functionality as `META-.`, will keep this open for `ESC-.` support, but the current implementation should help with the use case. Please enable the Left/Right Option as Meta in `Settings > Features > Keys`. Please test this feature and let us know if any issues.

https://github.com/warpdotdev/Warp/assets/16809145/b698ee99-307b-4ad9-888a-234aa0bb4fc7

**Comment by sladinji** [comment]
> Hey Folks, we've implemented this functionality as `META-.`, will keep this open for `ESC-.` support, but the current implementation should help with the use case. Please enable the Left/Right Option as Meta in `Settings > Features > Keys`. Please test this feature and let us know if any issues.
> CleanShot.2023-11-21.at.14.03.43.mp4

It does not work with my keyboard layout (FR). I tested with English layout and it works fine but unfortunately no way to make it works with my layout.

**Comment by huxiang** [comment]
> Hey Folks, we've implemented this functionality as `META-.`, will keep this open for `ESC-.` support, but the current implementation should help with the use case. Please enable the Left/Right Option as Meta in `Settings > Features > Keys`. Please test this feature and let us know if any issues.
> 
>  CleanShot.2023-11-21.at.14.03.43.mp4

`META-.` works for me, thanks for implementing it!

**Comment by jab** [comment]
Although `META-.` now works in the shell, it is still not working in other readline-enabled programs like the `python` or `ipython` REPLs, where it still results in `≥` being echoed rather than inserting the last word of the previous line.

**Comment by Zer0x00** [comment]
> > Hey Folks, we've implemented this functionality as `META-.`, will keep this open for `ESC-.` support, but the current implementation should help with the use case. Please enable the Left/Right Option as Meta in `Settings > Features > Keys`. Please test this feature and let us know if any issues.
> > CleanShot.2023-11-21.at.14.03.43.mp4
> 
> It does not work with my keyboard layout (FR). I tested with English layout and it works fine but unfortunately no way to make it work with my layout.

Same here with a German keyboard layout. Works in SSH but not in nano.

**Comment by ericmburgess** [comment]
> Hey Folks, we've implemented this functionality as `META-.`, will keep this open for `ESC-.` support, but the current implementation should help with the use case. Please enable the Left/Right Option as Meta in `Settings > Features > Keys`. Please test this feature and let us know if any issues.
> 
>  CleanShot.2023-11-21.at.14.03.43.mp4

Thank you for the attention, genuinely! But this is not nearly good enough. In a standard terminal, I can use `Esc-.` repeatedly to step backward through the last argument of previous commands. This is also why workarounds like using `$_` or `!$` don't cut it. I use it every bit as much as I use up-arrow, and it's murder living without it.

In the interest of making a concrete suggestion, can we not get a toggle to simply pass Esc through, unmodified, like every other terminal does?
